lib/ts/components/superTokensRoute.tsx

```diff
@@ -54,8 +54,10 @@ function SuperTokensRouteWithRecipeId({
     withRouter: WithRouterType;
 }): JSX.Element | null {
     const normalisedPath = new NormalisedURLPath(path);
-    const featureComponentWithRecipeId =
-        SuperTokens.getInstanceOrThrow().getMatchingComponentForRouteAndRecipeId(normalisedPath);
+
+    const featureComponentWithRecipeId = React.useMemo(() => {
```

Comment:
Two categories of issues with this:
1. Style: everywhere we use hooks like `useMemo(...)`, `useRef(...)` directly importing them as `import { useMemo, useRef } from "react"`. I would also stick to this style here.
2. Best practices
- `useMemo` is a solution to memoize a value of expensive computation, not just storing some value across rerenders (which `useRef` is for).
- `useMemo` shouldn't be relied on for behaviour of the component: https://reactjs.org/docs/hooks-reference.html#usememo

> You may rely on useMemo as a performance optimization, not as a semantic guarantee. In the future, React may choose to “forget” some previously memoized values and recalculate them on next render, e.g. to free memory for offscreen components. Write your code so that it still works without useMemo — and then add it to optimize performance.

I see that your concern is storing the instance unless normalisedPath changes. Does the normalisedPath change?

My feeling is we would be better here with class component, where we just store the instances.

Reply:
- About point number 1: It makes sense.
- Yea.. we can get rid of the `useMemo`.

> Does the normalisedPath change?

No

> My feeling is we would be better here with class component, where we just store the instances.

How would you store it? You can't store react components in a state, and idk if it's a good idea to store then in instance variables. I think the actual problem we need to solve here is that `SuperTokensRouteWithRecipeId` is loosing state.